### ISSUE TITLE
Problematic indent?

### ISSUE THREAD

**Issue by betaorbust** [nvuillam/vscode-groovy-lint#109]
I have the following code, which seems to be triggering some strange behaviors around the indent rule.
1) It's incorrectly flagging the indent on the return statement, but **only** in the VSCode plugin. The cli says no errors.
2) It's impossible to turn off. Even setting all rules to empty in both `.groovylintrc.json` and `groovylintrc-format.json` it still triggers.
```groovy
public class Test {
    public static Observable<Something> testing() {
        return api
            .flatMap({ Map something -> 
                if (!something) {
                    // This return is reported as being on the wrong level
                    return Observable.empty()
                }
            })
            // This next part is required to trigger the bug. If you delete it, the above return statement
            // is detected as being at the correct indent level 🤯 
            .flatMap()
    }
}
```

![Screen Shot 2021-04-08 at 12 09 59 AM](https://user-images.githubusercontent.com/921683/113983547-df5f8a80-97fe-11eb-9ac0-6d345515f8ff.png)
